src/Tokenizer/Tokens.php

```diff
@@ -1139,6 +1157,26 @@ public function clearTokenAndMergeSurroundingWhitespace($index)
     }
 
     /**
+     * Finds the line number of the first token previous to the one at the specified index, including itself.
+     *
+     * @param int $index Token index
+     *
+     * @return int Line number
+     */
+    private function findNearestPrevTokenLineNumber($index)
```

Comment:
method should not exist, as with your approach Token instance should have the line property always valid. here, you hack the code because you don't have it always provided. bad approach